### ISSUE TITLE
`make` fails in current vbox image + node/npm

### ISSUE THREAD

**Issue by jwhitley** [appuri/mock-http-server#2]
### Repro steps
1. `vagrant up`
2. `vagrant ssh`
3. `sudo apt-get install nodejs npm`
4. `cd /opt/src/mock-http-server`
5. `make`
### Actual results

```
$ make pristine && make
rm -f bin/*
rm -f lib/*.js
rm -f -r node_modules
npm install -d
npm info it worked if it ends with ok
npm info using npm@1.1.4
npm info using node@v0.6.12
npm info preinstall mock-http-server@1.0.0
npm http GET https://registry.npmjs.org/underscore/1.3.3
npm http GET https://registry.npmjs.org/optimist/0.3.4
npm http 200 https://registry.npmjs.org/underscore/1.3.3
npm http GET https://registry.npmjs.org/underscore/-/underscore-1.3.3.tgz
npm http 200 https://registry.npmjs.org/optimist/0.3.4
npm http GET https://registry.npmjs.org/optimist/-/optimist-0.3.4.tgz
npm http 200 https://registry.npmjs.org/underscore/-/underscore-1.3.3.tgz
npm http 200 https://registry.npmjs.org/optimist/-/optimist-0.3.4.tgz
npm info shasum 4d6d0bd71ffad0da4ba4f6d876d5eeb04e07480b
npm info shasum /tmp/npm-1337639050236/1337639050258-0.7120153564028442/tmp.tgz
npm info shasum 530615e7c571eade7ac37c98a2df6b7f109e7c4e
npm info shasum /home/vagrant/.npm/optimist/0.3.4/package.tgz
npm info shasum 47ac53683daf832bfa952e1774417da47817ae42
npm info shasum /tmp/npm-1337639050236/1337639050243-0.9576710190158337/tmp.tgz
npm info shasum 4756ad4182f434de5aae9b5e758c8ec6b12a6ae0
npm info shasum /home/vagrant/.npm/underscore/1.3.3/package.tgz
npm info into /opt/src/mock-http-server optimist@0.3.4
npm info into /opt/src/mock-http-server underscore@1.3.3
npm info installOne optimist@0.3.4
npm info installOne underscore@1.3.3
npm info unbuild /opt/src/mock-http-server/node_modules/optimist
npm info unbuild /opt/src/mock-http-server/node_modules/underscore
npm ERR! error installing optimist@0.3.4
npm info unbuild /opt/src/mock-http-server/node_modules/optimist
npm ERR! error rolling back optimist@0.3.4 Error: UNKNOWN, unknown error '/opt/src/mock-http-server/node_modules/optimist'

npm ERR! Error: ENOENT, no such file or directory '/opt/src/mock-http-server/node_modules/optimist/package.json'
npm ERR! You may report this log at:
npm ERR!     <http://bugs.debian.org/npm>
npm ERR! or use
npm ERR!     reportbug --attach /opt/src/mock-http-server/npm-debug.log npm
npm ERR! 
npm ERR! System Linux 3.2.0-23-generic
npm ERR! command "node" "/usr/bin/npm" "install" "-d"
npm ERR! cwd /opt/src/mock-http-server
npm ERR! node -v v0.6.12
npm ERR! npm -v 1.1.4
npm ERR! path /opt/src/mock-http-server/node_modules/optimist/package.json
npm ERR! code ENOENT
npm ERR! message ENOENT, no such file or directory '/opt/src/mock-http-server/node_modules/optimist/package.json'
npm ERR! errno {}
npm ERR! 
npm ERR! Additional logging details can be found in:
npm ERR!     /opt/src/mock-http-server/npm-debug.log
npm not ok
make: *** [node_modules] Error 1
```
### Expected results

make works as expected.  FWIW, this does work fine on OS X with node v0.6.17 and npm 1.1.21.

**Comment by ggoodale** [comment]
Can you try 'npm install optimist' by itself?  Does the npm-debug.log mentioned in the error have any additional detail?

**Comment by amolk** [comment]
Potentially VirtualBox shared folders related issue - https://github.com/isaacs/npm/issues/1375

**Comment by jwhitley** [comment]
Crud.  I know the VB team has done a ton of work on shared folders over recent release... apparently not enough.  Relevant bit of npm-debug.log:

```
verbose gunzed /opt/src/mock-http-server/node_modules/___optimist.npm/package
verbose rm'ed /opt/src/mock-http-server/node_modules/optimist
verbose renamed [ '/opt/src/mock-http-server/node_modules/___optimist.npm/package',
verbose renamed   '/opt/src/mock-http-server/node_modules/optimist' ]
ERR! error installing optimist@0.3.4
info unbuild /opt/src/mock-http-server/node_modules/optimist
ERR! error rolling back optimist@0.3.4 Error: UNKNOWN, unknown error '/opt/src/mock-http-server/node_modules/optimist'
```

**Comment by jwhitley** [comment]
This issue is now resolved by appuri/elite-vagrant#1.  The `make` workflow was deprecated by 64803aba167b37cd62104371b754627188f8a220, but that fix was still required to make the new `npm install -d` step work correctly.
